### PR TITLE
arch/imxrt/gpio: update gpio irqvector table

### PR DESCRIPTION
**Pull request by an4967**
- update irq vector table for all gpio set.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [035427d54bfc2608cd3bb7cb9d17b24873ff71b7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [035427d54bfc2608cd3bb7cb9d17b24873ff71b7] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_gpio.c [hunk 1]
```diff
@@ -734,10 +734,32 @@ uint8_t imxrt_gpio_irqvector(uint32_t pinset)
 	int port = (pinset & GPIO_PORT_MASK) >> GPIO_PORT_SHIFT;
 	int pin = (pinset & GPIO_PIN_MASK) >> GPIO_PIN_SHIFT;
 
-	if (port == GPIO_PORT1 >> GPIO_PORT_SHIFT) {
+	switch (port) {
+	case GPIO1INDEX:
+		if (pin > 15) {
+			return _IMXRT_GPIO1_16_31_BASE + (pin - 16);
```
Comment (manoharahk) [inline os/arch/arm/src/imxrt/imxrt_gpio.c]: IMO, It is good to use the switch statement to assign a value to a variable and break. Later return that variable after the switch statement at function exit.

Comment (an4967) [reply os/arch/arm/src/imxrt/imxrt_gpio.c]: I applied your comment. Thanks

**Comment by seinfra** [comment]
Target : [48adb64f0af7967b90d9ec9905679ad1cae8e690] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [48adb64f0af7967b90d9ec9905679ad1cae8e690] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/imxrt_gpio.c [hunk 2]
```diff
@@ -733,14 +733,48 @@ uint8_t imxrt_gpio_irqvector(uint32_t pinset)
...
+		} else {
+			ret = _IMXRT_GPIO5_0_15_BASE + pin;
+		}
+		break;
+	default:
+		break;
 	}
-
-	return 0;
+	return ret;
```
Comment (manoharahk) [inline os/arch/arm/src/imxrt/imxrt_gpio.c]: casting from int to uint8_t, are we loosing some bits here?
Total interrupts can be up-to (16 + 160 + (5 * 32)) = 336

**Comment by seinfra** [comment]
Target : [1d943c74693a0910999ccb11d135567e7f05d8b6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1d943c74693a0910999ccb11d135567e7f05d8b6] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/imxrt_gpio.c [hunk 3]
```diff
@@ -729,18 +729,52 @@ static inline int imxrt_gpio_configperiph(gpio_pinset_t pinset)
...
+	uint16_t ret = 0;
 
-	if (port == GPIO_PORT1 >> GPIO_PORT_SHIFT) {
-		return _IMXRT_GPIO1_0_15_BASE + pin;
-	} else if (port == GPIO_PORT2 >> GPIO_PORT_SHIFT) {
-		return _IMXRT_GPIO1_16_31_BASE + pin;
+	switch (port) {
+	case GPIO1INDEX:
+		if (pin > 15) {
+			ret = _IMXRT_GPIO1_16_31_BASE + (pin - 16);
```
Comment (Taejun-Kwon) [inline os/arch/arm/src/imxrt/imxrt_gpio.c]: We don't have to check CONFIG_IMXRT_GPIO1_0_15_IRQ & CONFIG_IMXRT_GPIO1_16_31_IRQ?
Base value is changed based on these config.

Comment (an4967) [reply os/arch/arm/src/imxrt/imxrt_gpio.c]: The base value in GPIO irq will be changed in previous step, imxrt102x_irq.h/imxrt105x_irg.h related to CONFIG_IMXRT_GPIO1_0_15_IRQ, ... CONFIG_IMXRT_GPIO5_16_31_IRQ

Comment (Taejun-Kwon) [reply os/arch/arm/src/imxrt/imxrt_gpio.c]: ok, got it!.

**Pull request merged**